src/node.rs

```diff
@@ -696,7 +696,7 @@ impl<K: Borrow<[u8]>, V> Node<K, V> {
                 // unsafe: same rationale.
                 if unsafe { root.as_ref().unchecked_unwrap().unwrap_branch_ref() }
                     
-                    .choice >= prefix.len()
+                    .choice <= prefix.len()
```

Comment:
```suggestion
                    .choice >= prefix.len() * 2
```

This check is here to make sure we don't remove a node which deals with sequences shorter than the prefix we're searching for. The bug here looks like we're comparing an index which is counting nybbles (the node's `choice` field) against an index counting bytes (`prefix.len()`) so on my machine, this fixes all tests while changing the direction of the comparison breaks a doctest. :)